BruteForce/BruteForce.cpp

```diff
@@ -1,20 +1,168 @@
+        reinterpret_cast<unsigned char*>(&password[0]),
+        password.size(), 1, key, iv))
+    {
+        throw std::runtime_error("EVP_BytesToKey failed");
+    }
 }
 
-// Запуск программы: CTRL+F5 или меню "Отладка" > "Запуск без отладки"
-// Отладка программы: F5 или меню "Отладка" > "Запустить отладку"
+void EncryptAes(const std::vector<unsigned char> plainText, std::vector<unsigned char>& chipherText)
+{
+    EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
+    if (!EVP_EncryptInit_ex(ctx, EVP_aes_128_cbc(), NULL, key, iv))
+    {
+        throw std::runtime_error("EncryptInit error");
+    }
+
+    std::vector<unsigned char> chipherTextBuf(plainText.size() + AES_BLOCK_SIZE);
+    int chipherTextSize = 0;
+    if (!EVP_EncryptUpdate(ctx, &chipherTextBuf[0], &chipherTextSize, &plainText[0], plainText.size())) {
+        EVP_CIPHER_CTX_free(ctx);
+        throw std::runtime_error("Encrypt error");
+    }
+
+    int lastPartLen = 0;
+    if (!EVP_EncryptFinal_ex(ctx, &chipherTextBuf[0] + chipherTextSize, &lastPartLen)) {
+        EVP_CIPHER_CTX_free(ctx);
+        throw std::runtime_error("EncryptFinal error");
+    }
+    chipherTextSize += lastPartLen;
+    chipherTextBuf.erase(chipherTextBuf.begin() + chipherTextSize, chipherTextBuf.end());
+
+    chipherText.swap(chipherTextBuf);
 
-// Советы по началу работы 
-//   1. В окне обозревателя решений можно добавлять файлы и управлять ими.
-//   2. В окне Team Explorer можно подключиться к системе управления версиями.
-//   3. В окне "Выходные данные" можно просматривать выходные данные сборки и другие сообщения.
-//   4. В окне "Список ошибок" можно просматривать ошибки.
-//   5. Последовательно выберите пункты меню "Проект" > "Добавить новый элемент", чтобы создать файлы кода, или "Проект" > "Добавить существующий элемент", чтобы добавить в проект существующие файлы кода.
-//   6. Чтобы снова открыть этот проект позже, выберите пункты меню "Файл" > "Открыть" > "Проект" и выберите SLN-файл.
+    EVP_CIPHER_CTX_free(ctx);
+}
+
+void CalculateHash(const std::vector<unsigned char>& data, std::vector<unsigned char>& hash)
+{
+    std::vector<unsigned char> hashTmp(SHA256_DIGEST_LENGTH);
+
+    SHA256_CTX sha256;
+    SHA256_Init(&sha256);
+    SHA256_Update(&sha256, &data[0], data.size());
+    SHA256_Final(&hashTmp[0], &sha256);
+
+    hash.swap(hashTmp);
+}
+
+void Encrypt()
+{
+    std::vector<unsigned char> plainText;
+    ReadFile("plain_text", plainText);
+
+    std::vector<unsigned char> hash;
+    CalculateHash(plainText, hash);
+
+    std::vector<unsigned char> chipherText;
+    EncryptAes(plainText, chipherText);
+
+    WriteFile("chipher_text", chipherText);
+
+    AppendToFile("chipher_text", hash);
+}
+std::vector<unsigned char> DecryptAes(const std::vector<unsigned char> chipherText )
+{
+    std::vector<unsigned char> plainText;
+    EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
+    EVP_DecryptInit_ex(ctx, EVP_aes_128_cbc(), NULL, key, iv);
```

Comment:
You should check the results of this function, EVP_DecryptUpdate and EVP_DecryptFinal_ex